services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,8 +223,6 @@ public void run()
             )
         );
         injector.getInstance(ExecutorLifecycle.class).join();
-        // Explicitly call lifecycle stop, dont rely on shutdown hook.
-        lifecycle.stop();
```

Comment:
I'm not sure what is the advantage of removing this and putting the requirement that each thread started by any lifecycle handler be a daemon thread , this assumption is hard to verify in unit tests and could cause peons hanging.
If the concern is that calling stop() twice might cause issues then we should probably use the `started` flag in the stop() method to ensure it is effectively a noop if called twice.